fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java

```diff
@@ -241,6 +299,161 @@ public long getVersion() {
+                        child2ParentMap.put(childName, parentName);
+                    }
+                }
+            }
+
+            // First processing counters by hierarchy
+            Map<String, TCounter> tCounterMap = node.counters.stream()
+                    .collect(Collectors.toMap(TCounter::getName, t -> t));
+            Queue<String> nameQueue = Lists.newLinkedList();
+            nameQueue.offer(ROOT_COUNTER);
+
+            while (!nameQueue.isEmpty()) {
+                String topName = nameQueue.poll();
+
+                if (!Objects.equals(ROOT_COUNTER, topName)) {
+                    TCounter tcounter = tCounterMap.get(topName);
+                    String parentName = child2ParentMap.get(topName);
+
+                    String target = topName;
+                    boolean updateSum = true;
+                    // For this kind of special counter, we need to merge them into another target
+                    // But since the value has been summarized, we only need to update MIN/MAX, but not SUM
+                    if (topName.startsWith(MERGED_INFO_PREFIX_MIN) || topName.startsWith(MERGED_INFO_PREFIX_MAX)) {
+                        updateSum = false;
+                        target = org.apache.commons.lang3.StringUtils.removeStart(target, MERGED_INFO_PREFIX_MIN);
+                        target = org.apache.commons.lang3.StringUtils.removeStart(target, MERGED_INFO_PREFIX_MAX);
+                    }
+
+                    SummarizationCounter summarization =
+                            getOrAddSummarizationCounter(target, tcounter.type, tcounter.strategy, parentName);
+                    if (summarization == null) {
+                        continue;
+                    }
+
+                    summarization.merge(tcounter, updateSum);
+                    tCounterMap.remove(topName);
+                }
+
+                if (node.child_counters_map != null) {
+                    Set<String> childNames = node.child_counters_map.get(topName);
+                    if (childNames != null) {
+                        for (String childName : childNames) {
+                            nameQueue.offer(childName);
+                        }
+                    }
+                }
+            }
+            // Second, processing the remaining counters, set ROOT_COUNTER as it's parent
+            for (TCounter tcounter : tCounterMap.values()) {
+                Pair<SummarizationCounter, String> pair = fragmentCounterMap.get(tcounter.name);
+                if (pair == null) {
+                    SummarizationCounter counter =
+                            addSummarizationCounter(tcounter.name, tcounter.type, tcounter.strategy, ROOT_COUNTER);
+                } else {
+                    Preconditions.checkArgument(pair.first.unit == tcounter.type);
+                    pair.first.merge(tcounter, true);
+                }
+            }
+
+            // Remove counters from this fragment-profile
```

Comment:
so after merge one instance profile into fragment profile, you actually merged counter and remove counters from instance profile, leave others same as before, right?

Reply:
yes. add a `purgeInstanceRunningProfile` to clarify this process